### PR TITLE
persist renderer disable through reload fix

### PR DESCRIPTION
**Pull request by Eriner**

### CODE REVIEW

**Comment by welwood08** [comment]
The logic in your change seems to be effectively the same as the old logic? I don't see how only this toggle fails after reload while all the other toggles work fine anyway.

What's actually broken is a line in toggleRenderer:
`w.g_Minigame.Renderer.renderer = trt_oldRender;`
should be:
`w.g_Minigame.Renderer.render = trt_oldRender;`

**Comment by Eriner** [comment]
Ah, you're correct. My commit fixes the case with the current code that caused the box to be checked, but after a reload, it animates. Let me look into this and I may re-pull. Thanks @welwood08

**Comment by Eriner** [comment]
@welwood08 it still requires the sanity check, but you're correct, that was a typo in the original code. I will add that fix to the pull request, but this fix is still valid, as it actually solves the problem (just correcting the typo does not).

**Comment by Eriner** [comment]
The typo fix allows the game to revert to the old renderer once the box is unchecked without a reload. The way the logic works, the default render function is only set to null once the else{} event is triggered, which requires user interaction in it's current state. By that logic, the game loads the original w.g_Minigame.Renderer.render function to it's true value, and therefor, without the check to see if it's disabled, there is no other function to set it to null.

tl;dr if manually checked/unchecked, it works. Code doesn't have catch function to set if disabled after reload. This fixes that as well as the typo which resulted in the renderer not being re-enabled after manual uncheck.

**Comment by Eriner** [comment]
This block:

```
    if (enableRenderer) {
        toggleRenderer();
    }    
```

The logic is, if the 'enbleRender' box is ticked, toggleRender(); which would cause the renderer to be enabled. But if the box was not ticked, this function is never hit, and the renderer stays to the default and is still enabled.

```
    if (enableRenderer) {
        toggleRenderer();
    }    
    if (!enableRenderer) {
        w.g_Minigame.Renderer.render = function() {}
    }    
```

Changing it to the above allows firstRun() to check if it is disabled (ticked), a condition which is otherwise not checked.

**Comment by welwood08** [comment]
I thought it would be something like that - I propose that the better solution is to remove all the if checks surrounding all the toggle calls since each of the toggle functions behaves correctly in the disable case.

**Comment by Eriner** [comment]
@welwood08 can you expand on your comment? I don't want to misunderstand you, and I'm not sure if I know what you mean. The best thing to to, imo, would to refactor the code to check for all of the checkboxes and act accordingly within firstRun().

**Comment by Eriner** [comment]
If/when I refactor, I will do it in a separate pull, I just want to get your input before I do.

**Comment by Eriner** [comment]
@welwood08 I've added you as a collaborator on my fork, so if you want to make any fixes, go for it.

**Comment by welwood08** [comment]
There are 3 toggle functions called inside firstRun: toggleCritText, toggleAllText and toggleRenderer, all surrounded by an if check. Having thought twice though, it's more consistent to leave the if checks in for now since not all the toggle functions behave correctly in the disable case.

I think it might be better to leave the call to toggleRenderer in firstRun where it was instead of replacing it with a duplicate of what it would do in the disable case. The bug was `if (enableRenderer)` should have been `if (!enableRenderer)`, which is a result of the slightly awkward inconsistent option names.

**Comment by Eriner** [comment]
ah, I see what you're saying. If it's unchecked, the default function is fine, but if it is checked (!enableRenderer) then it needs to be called. You're completely correct. I will amend that.

**Comment by Eriner** [comment]
Thanks @welwood08, that makes much more sense to read, and you were completely correct. The check for 
`if (enableRenderer)` was unneeded, as the default `w.g_Minigame.Renderer.render` function is fine in that case; no need to re-set it to what it already is.

**Comment by Eriner** [comment]
Okay @wchill I think this is ready for pull. It fixes the case where the game would not persist the state of the users renderer preference, and also fixes the case where if unchecked (after being previously checked and disabled) the renderer would not re-enable without a restart.

**Comment by Eriner** [comment]
Fixed through disableRenderer and toggleRenderer refactor. Closing.
